test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using System;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Add.Proj.Tests
+{
+    public class GivenDotnetAddProj : TestBase
+    {
+        [Theory]
+        [InlineData("--help")]
+        [InlineData("-h")]
+        public void WhenHelpOptionIsPassedItPrintsUsage(string helpArg)
+        {
+            var cmd = new DotnetCommand()
+                .ExecuteWithCapturedOutput($"add project {helpArg}");
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("idontexist.sln")]
+        [InlineData("ihave?inv@lid/char\\acters")]
```

Comment:
If one of these characters were accidentally treated as valid, then this test will hide the defect. Perhaps a separate `InlineData` per invalid character?